specification/resources/DeploymentStacks.Management/models-whatIf.tsp

```diff
@@ -0,0 +1,389 @@
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "Model is shared between resource group, subscription, and management group scope."
+  tags?: Record<string>;
+}
+
+alias DeploymentStacksWhatIfResultNameParameter = ResourceNameParameter<
+  Resource = DeploymentStacksWhatIfResult,
+  KeyName = "deploymentStacksWhatIfResultName",
+  SegmentName = "deploymentStacksWhatIfResults",
+  NamePattern = "^[-\\w\\._\\(\\)]+$"
+>;
+
+@@maxLength(DeploymentStacksWhatIfResult.name, 90);
+@@minLength(DeploymentStacksWhatIfResult.name, 1);
+@@doc(DeploymentStacksWhatIfResult.name,
+  "Name of the deployment stack what-if result."
+);
+
+/**
+ * DeploymentStack WhatIfResult Properties
+ */
+@added(Versions.v2025_07_01)
+model DeploymentStacksWhatIfResultProperties is DeploymentStacksError {
+  ...DeploymentStackSharedProperties;
+
+  /**
+   * The deployment stack id to use as the basis for comparison.
+   */
+  deploymentStackResourceId: string;
+
+  /**
+   * The timestamp for when the deployment stack was last modified. This can be used to determine if the what-if data is still current.
+   */
+  @visibility(Lifecycle.Read)
+  deploymentStackLastModified: utcDateTime;
+
+  /**
+   * The interval to persist the deployment stack what-if result in ISO 8601 format.
+   */
+  retentionInterval: duration;
+
+  /**
+   * All of the changes predicted by the deployment stack what-if operation.
+   */
+  @visibility(Lifecycle.Read)
+  changes?: DeploymentStacksWhatIfChange;
+
+  /**
+   * List of resource diagnostics detected by What-If operation.
+   */
+  @visibility(Lifecycle.Read)
+  @OpenAPI.extension("x-ms-identifiers", #[])
+  diagnostics?: DeploymentStacksDiagnostics[];
+}
+
+/**
+ * Changes predicted to the deployment stack as a result of the what-if operation.
+ */
+@added(Versions.v2025_07_01)
+model DeploymentStacksWhatIfChange {
+  /**
+   * List of resource changes predicted by What-If operation.
+   */
+  @OpenAPI.extension("x-ms-identifiers", #["identifiers", "resourceId"])
+  resourceChanges: DeploymentStacksWhatIfResourceChange[];
+
+  /**
+   * Predicted changes to the deployment stack deny settings.
+   */
+  denySettingsChange: DeploymentStacksChangeDelta<DenySettings>;
+
+  /**
+   * Predicted changes to the deployment scope for the deployment stack.
+   */
+  deploymentScopeChange: DeploymentStacksChangeBase<string>;
+}
+
+/**
+ * Information about a single resource change predicted by What-If operation.
+ */
+@added(Versions.v2025_07_01)
+model DeploymentStacksWhatIfResourceChange {
+  ...ResourceChangeDelta;
+
+  /**
+   * The resource id of the resource being changed.
+   */
+  resourceId: string;
+
+  /**
+   * The resource id of the Deployment responsible for this change.
+   */
+  deploymentId: string;
+
+  /**
+   * The symbolic name of the resource being changed.
+   */
+  symbolicName: string;
+
```

Comment:
Needs `resourceType?: string;` for extensible resource types. It is required as part of the composite ID for an extensible resource.

Reply:
Should now be covered as part of ResourceReference spread-ing